src/error.rs

```diff
@@ -82,6 +89,18 @@ pub enum Connect {
     ParseUrl(#[source] url::ParseError),
 }
 
+/// Errors happening client side
+#[derive(Debug, Error)]
+#[non_exhaustive]
+pub enum Client {
+    /// The 'Job' is malformed.
+    #[error("job is malformed: {desc}")]
+    MalformedJob {
```

Comment:
Do we ever expect to have other client-side errors than malformed jobs? I wonder if we should skip the intermediate `enum` here and instead have `MalformedJob` be a variant directly on `Error`.

Reply:
Looks like with other suggestions of yours applied, this variant is no longer needed (for now :))